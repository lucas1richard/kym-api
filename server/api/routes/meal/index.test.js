const { expect, use } = require('chai');
const { jestSnapshotPlugin } = require('mocha-chai-jest-snapshot');

describe('meal routes', () => {
  use(jestSnapshotPlugin());
  const routePrefix = '/api/meal';

  describe(`GET ${routePrefix}/v1`, () => {
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('gets public meals without keyword', async () => {
      const res = await globals.agent.getWithToken(`${routePrefix}/v1`);
      expect(res.statusCode).equal(200);
      expect(res.body).toMatchSnapshot();
    });

    it('gets public meals with keyword', async () => {
      const res = await globals.agent.getWithToken(`${routePrefix}/v1?keyword=chi`);
      expect(res.statusCode).equal(200);
      expect(res.body).toMatchSnapshot();
    });
  });
});
