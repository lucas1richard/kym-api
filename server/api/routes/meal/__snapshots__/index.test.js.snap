// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meal routes GET /api/meal/v1 gets public meals with keyword 1`] = `Array []`;

exports[`meal routes GET /api/meal/v1 gets public meals without keyword 1`] = `Array []`;
